dGame/dUtilities/SlashCommandHandler.cpp

```diff
@@ -840,23 +903,135 @@ void SlashCommandHandler::HandleChatCommand(const std::u16string& command, Entit
+				return;
+			}
+
+			count = std::stoi(args[2]);
+			
+			if (count < 1)
+			{
+				ChatPackets::SendSystemMessage(sysAddr, u"Cannot send less than 1 of an item. Defaulting to 1.");
+				count = 1;
+			}
+		}
+
 		uint64_t currentTime = time(NULL);
 		sql::PreparedStatement* ins = Database::CreatePreppedStmt("INSERT INTO `mail`(`sender_id`, `sender_name`, `receiver_id`, `receiver_name`, `time_sent`, `subject`, `body`, `attachment_id`, `attachment_lot`, `attachment_subkey`, `attachment_count`, `was_read`) VALUES (?,?,?,?,?,?,?,?,?,?,?,0)");
 		ins->setUInt(1, entity->GetObjectID());
-		ins->setString(2, "Darkflame Universe");
+		ins->setString(2, entity->GetCharacter()->GetMailSender());
 		ins->setUInt(3, receiverID);
 		ins->setString(4, playerName);
 		ins->setUInt64(5, currentTime);
-		ins->setString(6, "Lost item");
-		ins->setString(7, "This is a replacement item for one you lost.");
+		ins->setString(6, entity->GetCharacter()->GetMailSubject());
+		ins->setString(7, entity->GetCharacter()->GetMailBody());
 		ins->setUInt(8, 0);
 		ins->setInt(9, lot);
 		ins->setInt(10, 0);
-		ins->setInt(11, 1);
+		ins->setInt(11, count);
 		ins->execute();
 		delete ins;
+
+		// Prepare the output string.
+		std::stringstream ss;
+		ss << "Sending " << count << " of item " << lot << " to " << playerName << " with custom sender = '" << entity->GetCharacter()->GetMailSender() << "', subject = '" << entity->GetCharacter()->GetMailSubject() << "', and body '" << entity->GetCharacter()->GetMailBody() << "'.";
+		auto sysMessage = ss.str();
+
+		ChatPackets::SendSystemMessage(sysAddr, GeneralUtils::ASCIIToUTF16(sysMessage));
+
+		return;
+	}
+
+	else if (chatCommand == "mailitemall" && entity->GetGMLevel() >= GAME_MASTER_LEVEL_MODERATOR) {
+		// Display help for the user
+		if (args.size() < 2) {
+			ChatPackets::SendSystemMessage(sysAddr, u"Mails an item to all characters, or all online characters");
+			ChatPackets::SendSystemMessage(sysAddr, u"/mailitemall [all, online] <itemid> <count>");
+			return;
+		}
+
+		sql::PreparedStatement* stmt;
+		if (args[0] == "all") {
+			// Get a list of all characters on the server
+			stmt = Database::CreatePreppedStmt("select c.id, c.name from charinfo c;");
+		}
+		else if (args[0] == "online") {
+			// Get a list of all active characters on the server (based on activity_log)
+			stmt = Database::CreatePreppedStmt("select c.id, c.name from charinfo c, activity_log l where c.id = l.character_id and l.activity = 0 and l.id = (select max(ll.id) from activity_log ll where l.character_id = ll.character_id);");
+		}
+		else {
+			ChatPackets::SendSystemMessage(sysAddr, u"Invalid all/online argument. Please type 'all' or 'online'.");
+			return;
+		}
+
 		
-		ChatPackets::SendSystemMessage(sysAddr, u"Mail sent");
+		sql::ResultSet* res = stmt->executeQuery();
+
+		uint32_t lot;
+
+		// Validate the given LOT.
+		if (!GeneralUtils::TryParse(args[1], lot))
+		{
```

Comment:
Maybe I missed this the first time around, but this should be `if ( ... ) {`, e.g. the bracket on the same line.